tools/migrate-axis-bounds/migration.py

```diff
@@ -0,0 +1,53 @@
+from pathlib import Path
+import json
+import shutil
+import time
+import argparse
+
+# Traverses a binaryData directory and changes all datasource-properties.jsons
+# that include additionalAxes, increasing the upper bound by 1
+# This follows a change in the upper bound semantic to be exclusive
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--dry",
+        action="store_true",
+    )
+    parser.add_argument("binary_data_dir", type=Path, help="WEBKNOSSOS binary data dir")
+    args = parser.parse_args()
+    dry = args.dry
+    binary_data_dir = args.binary_data_dir
+
+    count = 0
+    print(f"Traversing datasets at {binary_data_dir.resolve()} ...")
+    for orga_dir in [item for item in binary_data_dir.iterdir() if item.is_dir()]:
+        for dataset_dir in [item for item in orga_dir.iterdir() if item.is_dir()]:
+            json_path = dataset_dir / "datasource-properties.json"
+            if json_path.exists():
+                changed = False
+                with open(json_path, 'r') as json_file:
+                    content = json.load(json_file)
+                    if "dataLayers" in content:
+                        for layer in content["dataLayers"]:
+                            if "additionalAxes" in layer:
+                                for axis in layer["additionalAxes"]:
```

Comment:
To avoid some nesting I would recommend something like `for layer in content.get("dataLayers", [])` for the two if/for combinations.